src/components/ColorPicker/components/AlphaField/tests/AlphaField.test.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import {mountWithAppProvider} from 'test-utilities/legacy';
+import {Spinner} from 'components';
+import {AlphaField} from '../AlphaField';
+import {TextField} from '../../../../TextField';
+
+describe('<AlphaField />', () => {
+  const mockProps = {
+    alpha: 0,
+    onChange: noop,
+  };
+
+  describe('alpha', () => {
+    it('converts alpha to percentage in the TextField', () => {
+      const alpha = 0.5;
+      const alphaField = mountWithAppProvider(
+        <AlphaField {...mockProps} alpha={alpha} />,
+      );
+      const textField = alphaField.find(TextField);
+      const expectedPercentage = Math.round(alpha * 100).toString();
+
+      expect(textField.prop('value')).toBe(expectedPercentage);
+    });
+  });
+
+  describe('onChange()', () => {
+    it('calls onChange with the alpha value when percentage is set', () => {
+      const onChangeSpy = jest.fn();
+      const alphaField = mountWithAppProvider(
+        <AlphaField {...mockProps} onChange={onChangeSpy} />,
+      );
+      const textField = alphaField.find(TextField);
+      const percentageValue = 70;
+      const expectedAlpha = percentageValue / 100;
+
+      (textField.find('input') as any).instance().value = percentageValue;
+      textField
+        .find('input')
+        .simulate('change')
+        .simulate('blur');
+
+      expect(onChangeSpy).toHaveBeenCalledWith(expectedAlpha);
+    });
+
+    it('calls onChange with alpha value of 1 when percentage value over 100 is set', () => {
+      const onChangeSpy = jest.fn();
+      const alphaField = mountWithAppProvider(
+        <AlphaField {...mockProps} onChange={onChangeSpy} />,
+      );
+      const textField = alphaField.find(TextField);
+      const percentageValue = 150;
+      const expectedAlpha = 1;
+
+      (textField.find('input') as any).instance().value = percentageValue;
+      textField
+        .find('input')
+        .simulate('change')
+        .simulate('blur');
```

Comment:
Why do we need to blur here, and in the below tests?

Reply:
`onChange` gets called by `handleBlur` [here](https://github.com/Shopify/polaris-react/pull/2186/files#diff-05e020bf01a1b218631cd41871d91d5dR42), so the `blur` event needs to occur for the `onChangeSpy` to be called. This is also the case with the other instances below.